### ISSUE TITLE
Blueman-applet pegging out CPU

### ISSUE THREAD

**Issue by Huckleberry582** [blueman-project/blueman#852]
blueman: blueman-applet 2.0.5
BlueZ:
Distribution: UbuntuMate 18.04
Desktop environment: Mate
I have a Dell Latitude E6500 that I am experimenting with Ubuntu 18.04. At start-up I was noticing the fan was running like crazy and then noticed the processor was pegged out. I found that if I turned off the wifi switch, this went away and the CPU idled down. Turning the switch back on ran it back up. Turning off Bluetooth on the applet caused it to go back to normal. Turning it back on ran it back up.

I updated Ubuntu this morning after installing it. Updates did not solve the issue. I followed the directions for debugging and pasted the output below. I turned it off after the initial output and you can see the results. Please let me know if there is anything else I can provide.

```
mike@ITLT-UBUNTU6500:~$ killall blueman-applet
mike@ITLT-UBUNTU6500:~$ blueman-applet --loglevel debug
blueman-applet version 2.0.5 starting
Stale PID, overwriting
_________
Load (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:60)
['StatusIcon', 'NetUsage', 'PPPSupport', 'DBusService', 'NMPANSupport', 'KillSwitch', 'GameControllerWakelock', 'RecentConns', 'AppIndicator', 'DiscvManager', 'AuthAgent', 'ShowConnected', 'PowerManager', 'NMDUNSupport', 'SerialManager', 'StandardItems', 'Menu', 'Headset', 'DhcpClient', 'TransferService', 'ExitItem', 'Networking'] 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/BlueZInterface.py:13)
Detected BlueZ 5 
/usr/lib/python3/dist-packages/blueman/plugins/applet/AppIndicator.py:8: PyGIWarning: AppIndicator3 was imported without specifying a version first. Use gi.require_version('AppIndicator3', '0.1') before import to ensure that the right version gets loaded.
  from gi.repository import AppIndicator3 as girAppIndicator
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StatusIcon.StatusIcon'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DBusService.DBusService'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading PPPSupport because it's conflict has higher priority 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMPANSupport.NMPANSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Menu.Menu'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.PowerManager.PowerManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.KillSwitch.KillSwitch'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.GameControllerWakelock.GameControllerWakelock'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.RecentConns.RecentConns'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AppIndicator.AppIndicator'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.DiscvManager.DiscvManager'> 
_________
update_menuitems (/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py:123)
warning: Adapter is None 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.AuthAgent.AuthAgent'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ShowConnected.ShowConnected'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.NMDUNSupport.NMDUNSupport'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.SerialManager.SerialManager'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.StandardItems.StandardItems'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Headset.Headset'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
Not loading DhcpClient because it's conflict has higher priority 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.TransferService.TransferService'> 
_________
get_interface_version (/usr/lib/python3/dist-packages/blueman/bluez/obex/Base.py:20)
Detected BlueZ integrated obexd 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.ExitItem.ExitItem'> 
_________
__load_plugin (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:133)
loading <class 'blueman.plugins.applet.Networking.Networking'> 
_________
load_nap_settings (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:36)
Loading NAP settings 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 2 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on True 
current state True 
new state False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
Signalling False 
_________
_on_obex_owner_changed (/usr/lib/python3/dist-packages/blueman/plugins/applet/TransferService.py:172)
obex owner changed: :1.54 
_________
on_dbus_name_owner_change (/usr/bin/blueman-applet:95)
org.bluez owner changed to :1.40 
_________
update_menuitems (/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py:123)
warning: Adapter is None 
_________
set_nap (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:65)
set nap False 
_________
enumerate_connections (/usr/lib/python3/dist-packages/blueman/plugins/applet/ShowConnected.py:53)
Found 0 existing connections 
_________
on_registered (/usr/lib/python3/dist-packages/blueman/bluez/obex/AgentManager.py:18)
/org/blueman/obex_agent 
_________
RequestPowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:131)
Requesting True 
_________
on_power_state_change_requested (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:118)
True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on False 
current state False 
new state False 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch changed 2 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on True 
current state False 
new state False 
_________
check (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:119)
callbacks done 
_________
set_adapter_state (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:90)
True 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch registered 4 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state False 
new state True 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
Signalling True 
_________
initialize (/usr/lib/python3/dist-packages/blueman/plugins/applet/RecentConns.py:144)
rebuilding menu 
_________
on_adapter_added (/usr/bin/blueman-applet:105)
Adapter added  /org/bluez/hci0 
_________
on_activate (/usr/bin/blueman-applet:108)
Adapter activated 
_________
initialize (/usr/lib/python3/dist-packages/blueman/plugins/applet/RecentConns.py:144)
rebuilding menu 
_________
register_agent (/usr/lib/python3/dist-packages/blueman/plugins/applet/AuthAgent.py:63)
Registering agent 
_________
enumerate_connections (/usr/lib/python3/dist-packages/blueman/plugins/applet/ShowConnected.py:53)
Found 0 existing connections 
_________
set_nap (/usr/lib/python3/dist-packages/blueman/plugins/applet/Networking.py:65)
set nap False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
RequestPowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:131)
Requesting False 
_________
on_power_state_change_requested (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:118)
False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off False 
foff False 
on True 
current state True 
new state True 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch changed 2 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on True 
current state True 
new state False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
Signalling False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on True 
current state False 
new state False 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch changed 4 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on True 
current state False 
new state False 
_________
check (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:119)
callbacks done 
_________
set_adapter_state (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:90)
False 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
killswitch removed 4 
_________
io_event (/usr/lib/python3/dist-packages/blueman/plugins/applet/KillSwitch.py:71)
State: False 
_________
UpdatePowerState (/usr/lib/python3/dist-packages/blueman/plugins/applet/PowerManager.py:159)
off True 
foff False 
on False 
current state False 
new state False 
_________
on_adapter_removed (/usr/bin/blueman-applet:115)
Adapter removed  /org/bluez/hci0 
_________
initialize (/usr/lib/python3/dist-packages/blueman/plugins/applet/RecentConns.py:144)
rebuilding menu 
_________
on_adapter_removed (/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py:90)
/org/bluez/hci0 
_________
Run (/usr/lib/python3/dist-packages/blueman/main/PluginManager.py:208)
Function on_adapter_removed on DiscvManager Failed 
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/blueman/main/PluginManager.py", line 212, in Run
    ret = getattr(inst, function)(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/blueman/plugins/applet/DiscvManager.py", line 92, in on_adapter_removed
    if path == self.adapter.get_object_path():
AttributeError: 'NoneType' object has no attribute 'get_object_path'
_________
enumerate_connections (/usr/lib/python3/dist-packages/blueman/plugins/applet/ShowConnected.py:53)
Found 0 existing connections 
mike@ITLT-UBUNTU6500:~$ 
```

* [ *] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

> ℹ Please use some service like [Pastebin](http://pastebin.com/) or [GitHub Gist](https://gist.github.com/) to post your logs and keep the thread clean and readable. Make sure to describe the exact steps you took when producing it.

**Comment by Huckleberry582** [comment]
By the way, I'm new to this and didn't see the part about posting in patebin or GitHub Gist. Please forgive this noob.

**Comment by Huckleberry582** [comment]
I gave a try to downloading the current master but when I tried to use gdebi to install it I was told it was the same as the currently installed package. I tried to reinstall it but was given this output:

dpkg: error: unable to read filedescriptor flags for <package status and progress file descriptor>: Bad file descriptor
I then tried using terminal to install because I read that sometimes gdebi might have issues with a file. I ran dpkg and then apt install with the -f argument per what i read about that error message. I then tested and it didn't correct the issue. Please let me know what else I might try. This is a computer I set up to try out 18.04 on and I am happy to experiment if it will help the community.

Mike

**Comment by infirit** [comment]
I installed the latest xubuntu 18.04 beta in a vm and I do not see the high cpu usage. If you open a task/process viewer is it blueman-applet that's at 100% or something else like bluetoothd?

I updated the installation docs for the git snapshots and hopefully this helps you try them out. Make sure you use the pythom 3.6 section.

**Comment by Huckleberry582** [comment]
Well, I may be in the wrong place. I assumed that the issue was with Bluetooth just because turning that off seemed to fix the CPU usage. I am attaching a screenshot so you can see what HTOP showed ast the offending process and what the system manager said.
![screenshot at 2018-03-28 16-54-56](https://user-images.githubusercontent.com/18648950/38056055-575d263c-32a9-11e8-919f-f710939642d5.png)
I'll have to look and see what systemd-udevd does and how it affects bluetooth.

Mike

**Comment by infirit** [comment]
udev manages device nodes in /dev when devices are added and removed. It may very well be that it is related to bluetooth just not blueman. You can use ``udevadm monitor`` to see what it is doing when you turn bluetooth on or off.

My suspicion is that udev runs a script or command that behaves badly.

**Comment by Huckleberry582** [comment]
I installed UbuntuMate 17.10 as a dual boot to see if it had the same issue and it didn't. But, as soon as I loaded up 18.04, it happened again. Somewhere along the way, however, something got messed up and the thing wouldn't boot either. I ended up just wiping it out and giving mx17.1 a try. It runs beautifully on this old laptop. I think I'll stick with it on this machine for a bit and wait a little while on trying 18.04 again.

Thanks for your help.
